CMakeLists.txt

```diff
@@ -40,33 +55,57 @@ target_include_directories(ghexlib INTERFACE
 target_link_libraries(ghexlib INTERFACE GridTools::gridtools MPI::MPI_CXX)
 target_compile_features(ghexlib INTERFACE cxx_std_14)
 
-include(FetchContent)
-FetchContent_Declare(
-  googletest
-  GIT_REPOSITORY https://github.com/google/googletest.git
-  GIT_TAG        release-1.10.0
-)
-FetchContent_GetProperties(googletest)
-if(NOT googletest_POPULATED)
-  FetchContent_Populate(googletest)
-  add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
-
-  # https://github.com/google/googletest/issues/2429
-  add_library(GTest::gtest ALIAS gtest)
-endif()
-
-add_library(gtest_main_mt utils/gtest_main.cpp)
-target_link_libraries(gtest_main_mt MPI::MPI_CXX GridTools::gridtools GTest::gtest)
-
-add_library(gtest_main_bench utils/gtest_main_bench.cpp )
-target_link_libraries(gtest_main_bench MPI::MPI_CXX GridTools::gridtools GTest::gtest)
-
-add_library(gtest_main_bench_mt ./utils/gtest_main_bench.cpp)
-target_link_libraries(gtest_main_bench_mt MPI::MPI_CXX GridTools::gridtools GTest::gtest)
-target_compile_definitions(gtest_main_bench_mt PRIVATE GHEX_BENCHMARKS_USE_MULTI_THREADED_MPI)
+enable_testing()
 
-add_subdirectory(tests)
-add_subdirectory(benchmarks)
+if (GHEX_BUILD_TESTS OR GHEX_BUILD_BENCHMARKS)
+  include(FetchContent)
+  FetchContent_Declare(
+    googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG        release-1.10.0
+  )
+  FetchContent_GetProperties(googletest)
+  if(NOT googletest_POPULATED)
+    FetchContent_Populate(googletest)
+    add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
+    # https://github.com/google/googletest/issues/2429
+    add_library(GTest::gtest ALIAS gtest)
+  endif()
+  
+  set(GTEST_LIB_SOURCES ${googletest_SOURCE_DIR}/googletest/src/gtest-all.cc)
+  set(GTEST_INCLUDE_DIRS ${googletest_SOURCE_DIR}/googletest/include ${googletest_SOURCE_DIR}/googletest/src ${googletest_SOURCE_DIR}/googletest/)
+  
+  if (GHEX_BUILD_TESTS)
+    add_library(gtest_main_mt ${GTEST_LIB_SOURCES} ./utils/gtest_main.cpp )
+    target_include_directories(gtest_main_mt PRIVATE ${CMAKE_SOURCE_DIR}/include ${GTEST_INCLUDE_DIRS})
```

Comment:
you shouldn't do this (link against `GTest::gtest`, see old code)

Reply:
This won't work - it segfaults on my machine. This is the only way so far I could get it to work